src/libraries/illink-sharedframework.targets

```diff
@@ -1,48 +1,63 @@
 <Project>
 
-  <Target Name="ILLinkTrimSharedFramework"
-          AfterTargets="Build"
+  <Target Name="GetSharedFrameworkAssembliesToTrim"
           DependsOnTargets="PrepareForAssembliesTrim">
 
-    <Message Text="Trimming $(PackageRID) runtime pack assemblies with ILLinker..." Importance="high" />
-
     <PropertyGroup>
-      <LibrariesTrimmedArtifactsPath>$([MSBuild]::NormalizePath('$(ILLinkTrimAssemblyArtifactsRootDir)', 'trimmed-runtimepack'))</LibrariesTrimmedArtifactsPath>
+      <SharedFrameworkAssembliesTrimmedArtifactsPath>$([MSBuild]::NormalizePath('$(ILLinkTrimAssemblyArtifactsRootDir)', 'trimmed-runtimepack'))</SharedFrameworkAssembliesTrimmedArtifactsPath>
     </PropertyGroup>
 
+    <ItemGroup>
+      <SharedFrameworkAssemblyToTrim Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+      <SharedFrameworkAssemblyToTrim Include="$(SystemPrivateCoreLibPath)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <SharedFrameworkAssemblyTrimmed Include="$(SharedFrameworkAssembliesTrimmedArtifactsPath)*.*" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="ILLinkTrimSharedFramework"
+          AfterTargets="Build"
+          DependsOnTargets="GetSharedFrameworkAssembliesToTrim;PrepareForAssembliesTrim"
+          Inputs="$(ILLinkTasksAssembly);@(SharedFrameworkAssemblyToTrim)"
+          Outputs="@(SharedFrameworkAssemblyTrimmed)">
+
+    <Message Text="Trimming $(PackageRID) shared framework assemblies with ILLinker..." Importance="high" />
+
     <PropertyGroup>
-      <RuntimePackILLinkArgs>$(ILLinkArgs)</RuntimePackILLinkArgs>
+      <SharedFrameworkILLinkArgs>$(ILLinkArgs)</SharedFrameworkILLinkArgs>
       <!-- update debug symbols -->
-      <RuntimePackILLinkArgs>$(RuntimePackILLinkArgs) -b true</RuntimePackILLinkArgs>
+      <SharedFrameworkILLinkArgs>$(SharedFrameworkILLinkArgs) -b true</SharedFrameworkILLinkArgs>
     </PropertyGroup>
 
     <ItemGroup>
-      <_LibrariesToTrim Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
-      <_LibrariesToTrim Include="$(SystemPrivateCoreLibPath)" />
-
-      <RootAssemblies Include="@(_LibrariesToTrim)">
+      <RootAssemblies Include="@(SharedFrameworkAssemblyToTrim)">
         <RootMode>library</RootMode>
       </RootAssemblies>
     </ItemGroup>
 
     <ItemGroup>
       <!-- Include suppression XML files bin-placed in earlier per-library linker run. -->
-      <_RuntimePackSuppressionsXmls Include="$(ILLinkTrimAssemblyRuntimePackSuppressionsXmlsDir)*.xml" />
+      <SharedFrameworkSuppressionsXml Include="$(ILLinkTrimAssemblyRuntimePackSuppressionsXmlsDir)*.xml" />
```

Comment:
The suppression XML is also input to this task

Reply:
Fixed, thanks.